extensions/interactions/LogicProof/LogicProof.js

```diff
@@ -109,33 +132,45 @@ oppia.directive('oppiaInteractiveLogicProof', [
             // We update the message only if the user has added or removed a
             // line break, so that it remains while they work on a single line.
             if (change.text.length > 1 || change.removed.length > 1) {
-              $scope.refreshMessages(editor);
+              $scope.checkForBasicErrors(editor);
             }
           });
         };
 
-        $scope.refreshMessages = function(editor) {
-          if($scope.mistakeMark) {
-            $scope.mistakeMark.clear();
+        $scope.checkForBasicErrors = function(editor) {
+          if (!$scope.messageIsSticky) {
+            $scope.clearMessage();
           }
           try {
             logicProofStudent.validateProof(
               $scope.proofString, $scope.questionInstance);
-            $scope.proofError = '';
           } catch(err) {
-            $scope.proofError = $scope.displayMessage(err.message, err.line);
-            $scope.mistakeMark = editor.doc.markText(
-              {line: err.line, ch: 0},
-              {line: err.line, ch: 100},
-              {className: 'logic-proof-erroneous-line'});
+            $scope.clearMessage();
+            $scope.showMessage(editor, err.message, err.line);
+            $scope.messageIsSticky = false;
           }
           // NOTE: this line is necessary to force angular to refresh the
-          // displayed proofError.
+          // displayed mistakeMessage.
           $scope.$apply();
         };
 
-        $scope.displayMessage = function(message, lineNumber) {
-          return 'line ' + (lineNumber + 1) + ': ' + message;
+        $scope.clearMessage = function() {
+          if($scope.mistakeMark) {
+            $scope.mistakeMark.clear();
+          }
+          $scope.mistakeMessage = '';
+        }
+
+        $scope.showMessage = function(editor, message, lineNum) {
+          $scope.mistakeMessage = $scope.renderMessage(message, lineNum);
+          $scope.mistakeMark = editor.doc.markText(
```

Comment:
Ah, I noticed that while playing, but did not realize it was indicative of an error at all. In fact I remember having no idea what the bolding represented, and just ignoring it.

Can you do some kind of red-background highlight instead, something like the example shown below?

![error-example](https://cloud.githubusercontent.com/assets/10575562/11920731/aa50a14a-a72b-11e5-936e-f8386f827853.png)